lib/models.ts

```diff
@@ -0,0 +1,62 @@
+export interface CDE {
+    VariableName: string;
+    PreciseCDEAbbrev: string;
+    PreciseTBICDETitle: string;
+    UnitOfMeasure: string;
+    Description: string;
+    DataType: string;
+    Comments: string;
+    MultipleValue: string;
+    PermittedValues: string;
+    MinimumValue: string;
+    MaximumValue: string;
+    InterLexID: string;
+}
+
+export interface Repository {
+    name: string;
+    fetch: (query: string) => Promise<CDE[]>;
+}
+
+export interface Config {
+    height: string; // in %
+    width: string; // in %
+}
+
+export interface InitParams {
+    cdeFileMapping: File;
+    labFileMapping: File | null;
+    datasetSample: File;
+    callback: (cdeFileMapping: File | null) => void;
+    repositories: Repository[];
+    config: Config;
+    labName: string
+}
+
+
+// Internal
+
+export enum CDEType {
+    CDE = 'CDE',
+    CustomDataDictionary = 'CustomDataDictionary'
+}
+
+export enum CDEStatus {
+    Mapped = 'mapped',
+    Unmapped = 'unmapped'
+}
+export interface DatasetCDEEntry {
+    type: CDEType;
+    status: CDEStatus;
+    // Include other optional string attributes as needed
+    [key: string]: string;
```

Comment:
we can get the properties to show in the custom dropdown component from this object keys.

![image](https://github.com/MetaCell/cde-mapper/assets/19196034/d9d5f787-0cad-4572-abe0-6a50b5132755)